File: rgctl-tests/ecommerce-python/app/schemas/product.py
from datetime import datetime

from pydantic import BaseModel, Field

from app.schemas.common import ORMModel


class CreateProductRequest(BaseModel):
    category_id: str
    name: str = Field(min_length=1, max_length=255)
    slug: str = Field(min_length=1, max_length=255)
    description: str = ""
    price_cents: int = Field(ge=0)
    stock: int = Field(ge=0, default=0)


class ProductResponse(ORMModel):
    id: str
    category_id: str
    name: str
    slug: str
    description: str
    price_cents: int
    stock: int
    created_at: datetime

File: rgctl-tests/ecommerce-rust/src/dto/product.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Deserialize)]
pub struct CreateProductRequest {
    pub category_id: String,
    pub name: String,
    pub slug: String,
    pub description: String,
    pub price_cents: i64,
    pub stock: i64,
}

#[derive(Debug, Serialize)]
pub struct ProductResponse {
    pub id: String,
    pub category_id: String,
    pub name: String,
    pub slug: String,
    pub description: String,
    pub price_cents: i64,
    pub stock: i64,
}
